src/classes/ALLO_Allocations_TDTM.cls

```diff
@@ -578,6 +581,7 @@ public class ALLO_Allocations_TDTM extends TDTM_Runnable {
                 //reparent the cloned allocation to the opportunity
                 for (Allocation__c allo : listAlloForInsert) {
                     Allocation__c alloInMap;
+                    Boolean isPercentOnly = false;
```

Comment:
Could this just be 
```suggestion
                    Boolean isPercentOnly = allo.Percent__c !=null;
```
and eliminate line 602?

Also, since the var is isPercent**Only**, does this need to check for `allo.Amount__c == null` too?

Reply:
I can add that as a fail safe, but you cannot enter both amount and percent from the UI.  Better to be safe, so I will add.